static/js/message_flags.js

```diff
@@ -113,6 +113,18 @@ exports.toggle_starred_and_update_server = function (message) {
     }
 };
 
+exports.unstar_all_messages = function () {
+    var starred_msg_ids = page_params.starred_messages;
+    _.each(starred_msg_ids, function (msg_id) {
+        var msg = message_store.get(msg_id);
+        msg.starred = false;
+        unread_ops.notify_server_message_read(msg);
```

Comment:
why are you marking these messages as read?  I think that only makes sense in the case that you clicked on the message to star it, not for this bulk operation.  Though realistically, these messages structurally cannot be unread anyway.

Reply:
Yeah I thought about that message_read thing. But I was not sure what exactly it does. I actually took reference from another function `toggle_starred_and_update_server`. So, I just implemented it in similar manner to that. I was sure, community will give reference regarding that.(i.e. to add it or not)

Reply:
In the future, please flag in your PR description things like this that you were unsure about -- it's an important part of submitting a review request to flag things in your PR that you are suspicious might be incorrect.

Reply:
Sure! I can see that, it could be dangerous, if one don't mention such things. Will do this for sure from next time.